### ISSUE TITLE
Refactor listview code

### ISSUE THREAD

**Issue by bendk** [pculture/unisubs#3457]
Once the project settings page is merged, I think we can do a few things to improve the listview code:

- Standardize the layout a bit
  - If the browser is wider than needed to display the data in the listview, divide the extra space equally between all columns, except the checkbox column and actions columns.
  - Make it so if the text for that column exceeds its width, then we use ellipsis to cut it off
  - Rework/improve the CSS to move the actions to the right
- Work on mobile/small width design
  - As the browser width shrinks, we should hide the less important columns.  Once we get to mobile, we should only show 1 or two columns
  - Allow touch-and-hold action to open the context menu on mobile
  - Maybe make things work on mobile devices that don't support CSS grid
- Simplify the CSS
  - Add CSS mixins to handle these cases without having to code things like `grid-template-columns`

**Comment by bendk** [comment]
For testing:
  - Test the listView elements on these pages:
    - Member directory (both the admin and non-admin version)
    - Video URL tab on the future UI page
    - Project settings page
  - Make sure that:
     - The lists look okay at all browser sizes (mobile, tablet, laptop, desktop)
     - For listviews with a context menu, long-press on mobile devices opens the context menu

**Comment by PCF-Testing** [comment]
@bendk: could you please merge dev into this branch? Thanks!

**Comment by PCF-Testing** [comment]
- [ ] On mobile browser size (under 750px width), list view loses checkboxes and (i) and vertical dots icons. It also becomes unresponsive to scrolling - it is not possible to scroll the page up or down

- [ ] On a real iPad, truncating long text in a column on http://unisubs.example.com:8000/en/styleguide/listview did not work - instead of being truncated, the text expanded to the right edge of the screen and spilled beyond it

- [ ] In tablet browser size, Change Role and Remove Member modals do not work - would not invoke on selecting a menu item

- [ ] On mobile width (under 750 px), project description column disappears from the table even though there is still plenty of space for it.

- [ ] On tablet width, Video URLs tab cannot accommodate the URLs' tags properly:
![screenshot from 2018-11-02 05 23 07](https://user-images.githubusercontent.com/288455/47915310-aa95bf00-de5f-11e8-8f2e-d33273642294.png)
On mobile width, tags and icons/menus completely disappear.

- [ ] When a column in the table is hidden due to lack of space, it would be really nice to let the user know what they are missing and how they can view all the available information on a different device / screen - in the current implementation, the user has no chance of knowing if they are missing any important info (such as the number of assignments)

**Comment by bendk** [comment]
> ￼On mobile browser size (under 750px width), list view loses checkboxes and (i) and vertical dots icons. It also becomes unresponsive to scrolling - it is not possible to scroll the page up or down

Losing columns is the point of the code.  I tried to remove the ones that seemed least useful to give the other columns more space.  The reason to remove the menu icon is because you can get the same thing on mobile with a long touch.  Also, we shouldn't even me showing those buttons (see below for more).
The reason to remove the checkboxes is that the actionBar doesn't work well on mobile.

I fixed the scrolling issue.

> ￼ In tablet browser size, Change Role and Remove Member modals do not work - would not invoke on selecting a menu item

This seems like it's the side effect of another bug: we should only be showing those buttons on mouse hover, not when you touch the row.  I changed it so the buttons don't show at all on mobile devices now (you can use long-press to get the menu, so the functionality is still there)

> ￼ On tablet width, Video URLs tab cannot accommodate the URLs' tags properly:

I hide that column on tablet size

> ￼ When a column in the table is hidden due to lack of space, it would be really nice to let the user know what they are missing and how they can view all the available information on a different device / screen - in the current implementation, the user has no chance of knowing if they are missing any important info (such as the number of assignments)

I don't disagree, but I'm not sure how to do that.  Could you open a new issue for this one, I think it's beyond the scope of this one.

**Comment by bendk** [comment]
￼ On a real iPad, truncating long text in a column on > http://unisubs.example.com:8000/en/styleguide/listview did not work - instead of being truncated, the text expanded to the right edge of the screen and spilled beyond it

This is working for me in crossbrowser testing with mobile safari 11.  I think that's the first version that supports `display: grid`, so it's the first version that we're going to support.

**Comment by PCF-Testing** [comment]
> Could you open a new issue for this one

https://github.com/pculture/amara-enterprise/issues/2199

- [ ] For video URLs, info tags and action controls (except "Add URL") are unavailable for all screen widths below ~1030px - is that acceptable? (Long press does not seem to do anything on a phone.)

- [ ] In Firefox, the menu icon on Members page and the upper table in /styleguide/listview/ is three horizontal dots instead of vertical
![listviewfirefoxhorizontaldots](https://user-images.githubusercontent.com/288455/48130291-089c1b00-e241-11e8-8b8a-50ffaa22e09b.png)


- [ ] .
>  I changed it so the buttons don't show at all on mobile devices now (you can use long-press to get the menu, so the functionality is still there)

Tried long press on a phone - it does invoke the context menu, but there is no option to show the hidden info, and when you click Remove or Change Role, nothing happens

- [ ] On Settings|Projects, there is no way to perform any editing on tablet and phone devices, except Add Project. Is this as expected?

**Comment by PCF-Testing** [comment]
Generally looks good. I am still seeing a couple of small issues affecting tablets and small laptops:

- [ ] Projects are not editable on tablets (or does long press work in this case as well? haven't found a way to simulate that on crossbrowsertesting)

- [ ] Video URLs not editable on small laptops (under 1030 px) and on tablets

Also, "+" icons on "Add Item" buttons do not show on the branch.

The rest looks fine to me.

**Comment by bendk** [comment]
I can do both of those with long-press.  I'm not sure how it works with crossbrowsertesting, but I tested it with my actual phone at some point and it worked.

I fixed the plus icon on the projects settings page.  Tell me if you see other places where it's messed up.

**Comment by PCF-Testing** [comment]
The plus icon looks good in all places now.

The long press works for me on a phone too. The previous comment was about tablets and laptops with small screen width. I don't have a way to check how a long press works on a tablet; sent a question to CBT support on whether this can be simulated.

**Comment by bendk** [comment]
I think laptops should see the buttons.  I'm pretty sure long-press will work on tablets and phones the same way.

Can we move this one to needs PR?  It seems like these last issues are not that critical and the current code is a definitely improvement over what we had before.

**Comment by PCF-Testing** [comment]
> I think laptops should see the buttons.
I am speaking about Video URLs tab, which has no buttons on screen width below 1030px.

Moving to Needs PR.

**Comment by PCF-Testing** [comment]
Deployed.
